docs/roles/developer/faq.md

```diff
@@ -37,24 +32,33 @@ We don't need GPU support as we are a POS chain and we require very little compu
 
 You can read more about our consensus strategy on our [Validator Quickstart](/docs/validator/staking-overview) and [Validator FAQ](/docs/validator/validator-faq).
 
-### If a dev writes a vulnerable dApp -- let's say it's vulnerable to SQL injection or has a double free or something doesn't work that put the validator's machine at risk -- is the validator implicitly taking on some risk there on behalf of the developer?
+### If a developer writes a vulnerable or malicious dApp, is a validator implicitly taking on risk?
 
 No. 
 
-We believe that we have handled the potential damages to the network on the protocol level. For example we have a lot of limiters that constrain how much data you can pass into a function call or how much compute you can do in one function call, etc. 
-
-That said, smart contract developers will need to be responsible for their own dApps. 
+We have handled the potential damages to the network on the protocol level. For example, we have a lot of limiters that constrain how much data you can pass into a function call or how much compute you can do in one function call, etc.
 
-Vulnerability there can still happen but the damage will be limited to the smart contract itself.
+That said, smart contract developers will need to be responsible for their own dApps, as there is no stage gate or approval process. All vulnerability can only damage the smart contract itself. Luckily, updating smart contracts is very smooth on NEAR, so vulnerabilities can be updated/patched to an account in ways that cannot be done on other blockchains.
 
-### When renting boxes from the major providers you can Dockerize an application and run it wherever you like without worrying about whether the machine has what you need (as long as the hardware is suitable). With this approach there's very little in terms of requirements for a node to be a validator so how are required libraries / other environmental dependencies for the dApp accounted for?
+### Can I use external libraries in my contract?
 
 Most libraries should still be usable.  However, we do have a size limit for compiled binary of a contract so it is possible that certain large libraries will not be compatible. 
 
 On the other hand, things like interaction with storage is done through our runtime API so it reduces a lot of effort on the backend side of things.
 
+### How do you update a property of an object within a PersistentVector?
```

Comment:
This and next question seems to be assemblyscript specific. Might be worth noting it.

Reply:
Noted, thanks